core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -970,7 +970,7 @@ sealed trait ZIO[-R, +E, +A]
    * Returns an effect whose success is mapped by the specified `f` function.
    */
   def map[B](f: A => B)(implicit trace: Trace): ZIO[R, E, B] =
-    flatMap(a => ZIO.succeed(f(a)))
```

Comment:
this will greatly reduce the instances of `ZIO.Sync`... should be interesting to see if this impacts runtime performance. Does FiberRuntime always prioritize `ZIO.flatMap`?

Reply:
At the moment, the FiberRuntime runloop checks first whether the effect is ZIO.Sync and then Exit. However, after all these changes to use Exit a lot more, I'm contemplating to move Exit to be the first one to be checked. WDUT?

Reply:
It's hard to say. I figure 90%+ of user code is probably not explicitly using Exit, but I could be wrong. I do think we should generally optimized for deeply nested ZIOs (via map/flatMap) as usercode is often not optimized to reduce that.

It would be cool if we could get stats on suspension depth and ask users to collect this information...